community/bolt/src/main/java/org/neo4j/bolt/security/auth/BasicAuthentication.java

```diff
@@ -70,9 +70,19 @@ public AuthenticationResult authenticate( Map<String,Object> authToken ) throws
         }
     }
 
+    @Override
+    public void logout( AccessMode accessMode )
+    {
+        if ( accessMode instanceof AuthSubject )
```

Comment:
That sounds nice, but I think currently BasicAuthentication have to deal with AccessMode.Static coming in (the session state is initialized with AccessMode.Static.NONE). 
I don't really like the looks of logout taking AccessMode, so another alternative I was considering is pulling up the AuthSubject into the session state instead of AccessMode.. What do you think?